ttnn/ttnn/device.py

```diff
@@ -72,4 +73,14 @@ def dump_device_memory_state(device, prefix=""):
     ttl.device.DumpDeviceMemoryState(device, prefix)
 
 
+def is_wormhole_b0():
+    ARCH_NAME = os.environ.get("ARCH_NAME", os.environ.get("TT_ARCH_NAME", "")).lower()
+    return "wormhole_b0" in ARCH_NAME
+
+
+def is_grayskull():
```

Comment:
Same here. Make this function to take in device and figure out the arch from the device.